nx/lib/nx/defn/expr.ex

```diff
@@ -635,12 +636,23 @@ defmodule Nx.Defn.Expr do
     if axis in axes do
       [{is, il} | merge_slice(axis + 1, axes, inner_start, start, inner_lengths, lengths)]
     else
-      [s | start] = start
+      [_ | start] = start
       [l | lengths] = lengths
-      [{is + s, l} | merge_slice(axis + 1, axes, inner_start, start, inner_lengths, lengths)]
+
+      [
+        {Nx.add(is, 1), l}
```

Comment:
I am not sure if this is correct, shouldn't it be `Nx.add(is, s)`? Maybe our tests pass with this but we are lacking coverage...